cmd/server-mux.go

```diff
@@ -113,30 +113,46 @@ func (c *ConnMux) PeekProtocol() (string, error) {
 	return protocolTLS, nil
 }
 
-// Read - streams the ConnMux buffer when reset flag is activated, otherwise
-// streams from the incoming network connection
-func (c *ConnMux) Read(b []byte) (n int, e error) {
-	// Push read deadline
-	c.Conn.SetReadDeadline(time.Now().Add(defaultTCPReadTimeout))
-
-	// Update server's connection statistics
-	defer func() {
-		globalConnStats.incInputBytes(n)
-	}()
+// Read reads from the tcp session for data sent by
+// the client, additionally sets deadline for 15 secs
+// after each successful read. Deadline cancels and
+// returns error if the client does not send any
+// data in 15 secs. Also keeps track of the total
+// bytes received from the client.
+func (c *ConnMux) Read(b []byte) (n int, err error) {
+	n, err = c.peeker.Read(b)
+	if err != nil {
+		return n, err
+	}
+
+	// Read deadline was already set previously, set again
+	// after a successful read operation for future read
+	// operations.
+	c.Conn.SetReadDeadline(UTCNow().Add(defaultTCPReadTimeout))
+
+	// Update total incoming number of bytes.
+	globalConnStats.incInputBytes(n)
```

Comment:
In my understanding that is how it was even now with current master. It only makes sense to track success. Otherwise this would give misleading values since we need to track error separately.